### PR TITLE
Sidebar, footer and header fix, sidebar-main content display JS interaction, responsiveness fix and minor visual changes to the sections respectively

### PR DESCRIPTION
**Pull request by DimitarIM**
The sidebar might look weird on inspection with responsive dimensions ON. I have no idea why, don't want to even know. Rob said it's ok so I'm gucci. Try and test it with specific devices and tell me if anything's wrong. 

////////////////////////////////////////////////////

I also decided to integrate the homepage-sidebar-JS pull request with this one. Just for conveniency's sake. I'll finally close the other one and deem it merged. 

///////////////////////////////////////////////////

The general.css file that was filled with unnecessary code in main was also cleaned.

### CODE REVIEW

File: scripts/index.js [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+                <span class="tooltip">Kookaburra</span>
+            </li>
+            <li>
+                <a href="#" class="animal">
+                    <img src="./icons/cockatoo.png" alt="yellow-tailed-black-cockatoo Icon"
+                        class="icon-side eyellow-tailed-black-cockatoo-icon">
+                    <span class="nav-item">Yellow-Tailed Black Cockatoo</span>
+                </a>
+                <span class="tooltip">Yellow-Tailed Black Cockatoo</span>
+            </li>
```
Comment (Alexander-Lindstroem) [inline scripts/index.js]: You should use a loop to generate the sidebar, employing the objects in the array.

**Review by Alexander-Lindstroem — CHANGES_REQUESTED** [review]

File: scripts/index.js (hunk 1 above)
Comment (DimitarIM) [reply scripts/index.js]: Fixed it

File: scripts/index.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    sidebarHTML.innerHTML += `
+            <li>
+                <a href="#" class="animal">
+                    <img src="./icons/${animal.name.toLowerCase()}.png" alt="${animal.name.toLowerCase()} Icon" class="icon-side ${animal.name.toLowerCase()}-icon">
+                    <span class="nav-item">${animal.name}</span>
+                </a>
+                <span class="tooltip">${animal.name}</span>
+            </li>
+    `
+});
```
Comment (Alexander-Lindstroem) [inline scripts/index.js:186]: Thumbs up

**Review by Alexander-Lindstroem — APPROVED** [review]

**Review by AwdamnKaya — COMMENTED** [review]
LGTM

**Review by AwdamnKaya — APPROVED** [review]
LGTM still

**Pull request merged**